machine/v1beta1/types_azureprovider.go

```diff
@@ -174,6 +174,13 @@ type Image struct {
 	Version string `json:"version"`
 	// ResourceID specifies an image to use by ID
 	ResourceID string `json:"resourceID"`
+	// ThirdPartyImage indicates the image is published by a third party publisher and a Plan
+	// will be generated for it.
+	// +optional
+	ThirdPartyImage bool `json:"thirdPartyImage,omitempty"`
+	// SharedGallery specifies an image to use from an Azure Shared Image Gallery
```

Comment:
Why would I want to use this?  What image is it overriding if present?

Reply:
If you use this, you are specifying you are using an image from a Shared Gallery (rather than the Marketplace or by specifying an image ID, which is our current approach for installs). 

If you are using a Shared Gallery image, `publisher`, `sku`, & `offer` from the parent struct will be used to populate a required purchase plan. The fields from this struct would be used to create a ref to the image.